### PR TITLE
Handle invalid pixel IDs in LoadEventNexus

### LINKED ISSUE #19103: Unexpected pixel ID in `LoadEventNexus` causes out-of-range access
This was found while working on #18829 

### Expected behavior
If `LoadEventNexus` encounters unexpected pixel IDs, it should handle them elegantly, giving a warning if necessary if the problem is with the input file.

### Actual behavior
Unexpected pixel IDs cause an out-of-range vector access.

### Steps to reproduce the behavior
Run this script in a Debug version of mantid:
```
run=Load(Filename='OFFSPEC39347')
```
You should see a debug assertion.

### Platforms affected
Windows 7 and 10, maybe all

### PR DESCRIPTION
**Pull request by gemmaguest**
This fixes a debug assertion caused by invalid pixel IDs in an event file.

**To test:**

Run LoadEventNexusTest in Debug mode. Previously this caused a debug assertion. It should now be fixed.

Fixes #19103 .

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/ProcessBankData.cpp [hunk 1]
```diff
@@ -242,5 +242,20 @@ void ProcessBankData::run() { // override {
 #endif
 } // END-OF-RUN()
 
+// Get the workspace index for a given pixel ID. Throws if the pixel ID is
+// not in the expected range.
+size_t ProcessBankData::getWorkspaceIndexFromPixelID(const detid_t pixID) {
```
Comment (PranavBahuguna) [inline Framework/DataHandling/src/ProcessBankData.cpp]: You should add the '@param' and '@return' tags to the method description for parameters and return respectively.

**Comment by PranavBahuguna** [comment]
Code looks good and tests pass as expected. All that needs to be done is to address the line comment I made and add relevant labels and milestone to this PR.

**Comment by peterfpeterson** [comment]
I tried this out on `SNAP_36333` (the IDF says they have a pixel 0) and it appears to work. I need to search more for files that actually have counts in pixel 0 though.

**Comment by peterfpeterson** [comment]
There are only a couple of instruments that have pixel 0, and it always appears to occur in a dead area of the detector. I think this can be merged without issue for SNS.

**Comment by PranavBahuguna** [comment]
:shipit:

**Pull request merged**